### PR TITLE
feat: adds python SDK

### PR DESCRIPTION
**Pull request by MelvinKl**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest]
         go-version: ["1.18", "1.19", "1.20", "1.21", "1.22"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
```
Comment (vicentepinto98) [inline .github/workflows/ci.yaml:16]: please split the python CI into a separate job, it should be run in parallel with the Go CI

See https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-jobs-in-a-workflow

File: .github/workflows/sdk-pr.yaml [hunk 2]
```diff
@@ -31,11 +31,19 @@ jobs:
...
+        uses: ./.github/actions/generate-sdk/go
+      - name: Push Go SDK
         env: 
           GH_REPO: 'stackitcloud/stackit-sdk-go'
           GH_TOKEN: ${{ secrets.SDK_PR_TOKEN }}
         run: |
-          scripts/sdk-create-pr.sh "generator-bot-${{ github.run_id }}" "Generated from GitHub run [${{ github.run_id }}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})"
\ No newline at end of file
+          scripts/sdk-create-pr.sh "generator-bot-${{ github.run_id }}" "Generated from GitHub run [${{ github.run_id }}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})"
+      - name: Generate Python SDK
```
Comment (vicentepinto98) [inline .github/workflows/sdk-pr.yaml:42]: This should also be a separate job

File: scripts/generate-sdk/languages/python.sh [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            --input-spec "${service_json}" \
+            --output "${SERVICES_FOLDER}/${service}" \
+            --package-name "stackit.${service}" \
+            --template-dir "${ROOT_DIR}/templates/python/" \
+            --git-host ${GIT_HOST} \
+            --git-user-id ${GIT_USER_ID} \
+            --git-repo-id ${GIT_REPO_ID} \
+            --global-property apis,models,modelTests=false,modelDocs=false,apiDocs=false,apiTests=false,supportingFiles \
+            --additional-properties=pythonPackageName="stackit-${service},removeEnumValuePrefix=false" >/dev/null
+        ##--enable-post-process-file \
```
Comment (vicentepinto98) [inline scripts/generate-sdk/languages/python.sh:110]: remove the line instead of commenting out

File: scripts/project.sh [hunk 4]
```diff
@@ -17,6 +17,7 @@ elif [ "$action" = "tools" ]; then
     cd ${ROOT_DIR}
 
     go install golang.org/x/tools/cmd/goimports@latest
+    pip install black==24.8.0 isort~=5.13.2 autoimport~=1.6.1
```
Comment (vicentepinto98) [inline scripts/project.sh:20]: :+1:

File: scripts/sdk-create-pr.sh [hunk 5]
```diff
@@ -33,6 +33,13 @@ else
     REPO_URL_SSH=$3
 fi
 
+if [[ -z $4 ]]; then
+    echo "LANGUAGE not specified, default will be used."
+    LANGUAGE="go"
+else
+    LANGUAGE="python"
```
Comment (vicentepinto98) [inline scripts/sdk-create-pr.sh:40]:
```suggestion
    LANGUAGE=$4
```

File: templates/python/README.mustache [hunk 6]
```diff
@@ -0,0 +1,60 @@
+# {{{projectName}}}
```
Comment (vicentepinto98) [inline templates/python/README.mustache:1]: the templates already include the `oneOf` fixes or are you adding that later?

File: scripts/generate-sdk/languages/python.sh (hunk 3 above)
Comment (MelvinKl) [reply scripts/generate-sdk/languages/python.sh]: done

File: .github/workflows/sdk-pr.yaml (hunk 2 above)
Comment (vicentepinto98) [reply .github/workflows/sdk-pr.yaml:42]: you didn't separate this into two jobs, why? I think the python SDK generation should not fail if Go fails, they are independent

Comment (MelvinKl) [reply .github/workflows/sdk-pr.yaml]: You are correct. I split it

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**